aiohttp/client_proto.py

```diff
@@ -234,14 +271,22 @@ def data_received(self, data: bytes) -> None:
                 # parse http messages
                 try:
                     messages, upgraded, tail = self._parser.feed_data(data)
-                except BaseException as exc:
+                except BaseException as underlying_exc:
                     if self.transport is not None:
                         # connection.release() could be called BEFORE
                         # data_received(), the transport is already
                         # closed in this case
                         self.transport.close()
                     # should_close is True after the call
-                    self.set_exception(exc)
+                    reraised_exc: BaseException = ClientPayloadError(
+                        f"Unable to parse response payload: {underlying_exc !r}"
+                    )
+                    if isinstance(
+                        underlying_exc,
+                        (ClientError, HttpProcessingError),
+                    ):
+                        reraised_exc = underlying_exc
```

Comment:
```suggestion
                    reraised_exc = underlying_exc
                    if not isinstance(
                        underlying_exc,
                        (ClientError, HttpProcessingError),
                    ):
                        reraised_exc = ClientPayloadError(
                            f"Unable to parse response payload: {underlying_exc !r}"
                        )
```

trivial: I'd probably reverse this so it only reconstructs in the case where it doesn't match

Reply:
Fair. I tend to want less indentation so I optimize for readability. Though, your variant seems more efficient.